modin/engines/base/frame/data.py

```diff
@@ -1642,6 +1642,73 @@ def broadcast_apply_full_axis(
             validate_axes="all" if new_partitions.size != 0 else False,
         )
 
+    def _apply_full_axis_to_another(
```

Comment:
this also looks like a full duplication of [`broadcast_apply_full_axis`](https://github.com/modin-project/modin/blob/2b0b755753988cefbab38a71db973a26cbe581f4/modin/engines/base/frame/data.py#L1583), I think we can make a common method that they both will use